vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -764,12 +755,15 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
         });
 
         grid.$connector.removeFromQueue = tryCatchWrapper(function (item) {
-          let itemId = grid.getItemId(item);
-          // The treePageCallbacks for the itemId are about to be discarded ->
+          // The page callbacks for the given item are about to be discarded ->
           // Resolve the callbacks with an empty array to not leave grid in loading state
-          Object.values(treePageCallbacks[itemId] || {}).forEach((callback) => callback([]));
+          const itemContext = dataProviderController.getItemContext(item);
+          const itemSubCache = itemContext?.subCache;
+          if (itemSubCache) {
+            Object.values(itemSubCache.pendingRequests).forEach((callback) => callback([]));
+          }
```

Comment:
nit: this could still be a one-liner with `Object.values(itemContext?.subCache?.pendingRequests || {}).forEach((callback) => callback([]));`. Ignore this suggestion if you prefer the readability of the current change.

Reply:
Done.